pkg/types/query/primitives/primitives.go

```diff
@@ -50,22 +50,21 @@ func (f *Block) Accept(visitor Visitor) {
 	visitor.Block(*f)
 }
 
-type ConfirmationLevel int32
+type ConfidenceLevel float64
 
-// TODO come up with a proper chain agnostic solution BCF-3156
 const (
-	Finalized   = ConfirmationLevel(-1)
-	Unconfirmed = ConfirmationLevel(-2)
+	Highest = ConfidenceLevel(1)
```

Comment:
I noticed that the most often used values for confirmations are (in block confirmations):
* 0 block confirmations
* finalized (based on finality depth or tag)
* 1 block confirmation

I believe that 0 block confirmations and finalized are very easy to model with that approach - it would be Lowest and Highet, but how about 1 block confirmation? AFAIK in CCIP we don't use that, but I saw some core components relying on 1 block confirmation, e.g. config_poller, contract_transmitter

Reply:
The case of a 1 block confirmation would be a configuration in the chain reader somewhere on the spectrum of `[0..1]` in confidence level. The goal of switching to confidence level is to decouple the concept of finality and confirmations from the abstract confidence of being able to act on any particular event/log on any chain without reorgs or the like.

Therefore, the mappings of example confidence levels like `Medium`, or `KindaHigh` might map to very different concepts such as actual confirmations or a strict finality tag. Each chain can do something different AND each contract/event can be mapped different, allowing products a bit more freedom.

Reply:
I still have some reservations about making it a continuous parameter. I like the idea of letting people configure different custom levels like `Medium` or `KindaHigh`, but the case of 1-block confirmations is a good example of where a continuous confidence level doesn't make a whole lot of sense.

You could define a confidence level called `OneBlock` either way I guess. But if it were up to me, I'd avoid giving people the misleading impression that this is an actual "confidence" value in the statistical sense (a p-value between 0 and 1, representing Bayesian likelihood).  We could just have it be an integer between 0 and 10... very unlikely anyone would need to define more than 3 or 4 different levels... so even allowing 10 different values gives us plenty more flexibility than we'll ever need.

Reply:
Oh nevermind... I read this comment before looking over the code. Looks like you have already changed it to an enum, great!

Reply:
Yup, this was changed to enums. I'll close this conversation.